### DISCUSSION TITLE
Shubham Rawal- Github Externship Proposal

### DISCUSSION THREAD

**Discussion by officalshubhamrawal** [hoppscotch/hoppscotch#1651] [Q&A]
**Application number:**  18-05_Shu11_tfa_258

**Full name:**  Shubham Rawal

**E-mail:** shubham_rawal.scsebtech@galgotiasuniversity.edu.in

**GitHub username:**
@officalshubhamrawal

**Prior experience:**
- [x] HTML
- [x] CSS
- [x] SCSS
- [ ] Tailwind CSS
- [x] JavaScript
- [ ] TypeScript
- [x] Vue.js
- [ ] Nuxt.js
- [ ] NestJS
- [x] Firebase
- [ ] Prisma
- [ ] GraphQL
- [ ] PostgreSQL
- [ ] Go lang

---

**Proposal type:**
- [ ] Existing feature request
- [x] New feature proposal
- [ ] Refactor existing feature

**Proposal title:**  Synthetically Generate Data for Speed Testing 

**Description:**

It's difficult to test out API speed on a large data set or with real-time data with n rows and different types.
We can boost data-driven testing by creating synthetic data like addresses, security numbers and more or we can import the data from a CSV file. Users can create the fields for the number of rows they want the data should have, what type it should be in. etc.

This will help us test out the Speed of API on various real-time datasets or even some business-specific datasets.

**Proposal roadmap:** 
1. Wireframing and designing the UI/UX
2. Implementing User Side with dashboard
3. Implementing Backend with logic
4. implementing MockServer for rapid testing
5. implementing cross platform support for 3rd party integrations

**Comment by AndrewBastin** [comment]
Could you expand on this a bit more, your proposal seems to be a bit vague and unfocused.
